### ISSUE TITLE
Messenger as new channel

### ISSUE THREAD

**Issue by kuzmany** [mautic/mautic#4159]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | 
| Feature request? | y
| Enhancement? | 

## Description:
This is the biggest opportunity to attract attention for new Mautic clooud customers.
Messenger is largest cross platform chat service (1 bil users).

https://developers.facebook.com/docs/messenger-platform 

The integration (easiest way):

1. Visitors could confirm receive information to Messenger  (similar like Add to newsletter ). 
Integration should be with plugin https://developers.facebook.com/docs/messenger-platform/plugin-reference/checkbox-plugin
2. Mautic could send message to confirmed users via campaign action

Symfony libraries:

https://github.com/FriendsOfSymfony/FOSMessageBundle 
https://github.com/tgallice/fb-messenger-sdk

**Comment by luizeof** [comment]
@kuzmany this is a great feature. Do you will open on Mautic Founds on website??

**Comment by leonardoborlot** [comment]
+1 good feature, this will turn the income to Mautic process more short and easy!

**Comment by kuzmany** [comment]
@luizeof let's fund :) https://mautic-fund.kuzmany.biz/campaign/6/messenger-as-new-channel

**Comment by TomRoethlisberger** [comment]
+1 I like. Let's go for it. I will check out the funding page and donate

**Comment by kuzmany** [comment]
@TomRoethlisberger Thanks. I've already started work on this plugin.

**Comment by paulocastellano** [comment]
+1

**Comment by jimoquinn** [comment]
@kuzmany I donated, but I want to make a deal with you.  

I believe the description on both the donation page and this page is lacking.  What is the value proposition in layman's terms?  Can you create screen shots of Mautic and FB Messenger showing how it will benefit Mautic users?  Maybe provide an outline of the work required so people will understand the basic scope of the task?  

You do that and I'll promote the funding page across Mautic forums and slack.  Deal?

**Comment by kuzmany** [comment]
@jimoquinn plugin should add checkbox plugin to mautic forms or use checkbox plugin to website (checkout process for example) more https://developers.facebook.com/docs/messenger-platform/plugin-reference/checkbox-plugin
Other campaigns actions will works similar like notification - the same logic

**Comment by kuzmany** [comment]
30% done
Added screenshots to project
https://mautic-fund.kuzmany.biz/campaign/6/messenger-as-new-channel

**Comment by jimoquinn** [comment]
I posted a teaser on the forum using your information from the above posts.  Those who express interest will be asked to donate and I will clarify that I'm not the developer, but Zdeno Kuzmány is the developer.  He is responsible for numerous patches and new features for Mautic.

If not one person donates, I may haze them a bit (lol) for taking so much from the community and not giving back.  No code, no PR testing, and now too lazy to donate money.

**Comment by uzletiblog** [comment]
@kuzmany Do you know chatfuel.com?

It is a free chatbot builder platform. It has api to broadcast: https://help.chatfuel.com/facebook-messenger/broadcasting/broadcasting-api/

Check it out. Maybe you can use it and make your life easier

**Comment by TomRoethlisberger** [comment]
Hi @kuzmany How are you progressing with the messenger plugin? If you need help from our side, we would be gladly of assistance :-)

**Comment by kuzmany** [comment]
@TomRoethlisberger 
now, it's low priority :( 
but everbody can contribute to https://github.com/mautic/mautic/pull/4449 or donate another hours of work

**Comment by arthurpadilha** [comment]
@kuzmany what happened to the project? No interest on continuing it? This is an awesome feature

**Comment by vexersa** [comment]
I think FB's new terms and conditions may have halted this.

**Comment by arthurpadilha** [comment]
Damn, it would be very nice to have it on Mautic. Thank @vexersa

**Comment by dgram** [comment]
i think it will be really cool to integrate something like https://github.com/botman/botman to handle more than the facebook messenger. there are many communities interact with https://telegram.org also.
